tests/app/dao/test_fact_notification_status_dao.py

```diff
@@ -338,6 +340,32 @@ def test_get_last_send_for_api_key(notify_db_session):
     assert abs(time_delta.total_seconds()) < 1
 
 
+def test_get_api_key_ranked_by_notifications_created(notify_db_session):
```

Comment:
My tests for the query vs the endpoint are not radically different - mostly cut and pasted but the returned data is in a different shape. Not sure if I should be trying to make these tests more different. The query tests seem a bit redundant since the queries are being run and the returned data checked as part of the endpoint tests.